service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingBookingService.java

```diff
@@ -74,10 +80,60 @@ public HearingBooking getHearingBookingByUUID(List<Element<HearingBooking>> hear
         return combinedHearingDetails;
     }
 
+    public List<Element<HearingBooking>> setHearingJudge(List<Element<HearingBooking>> hearingBookings,
+                                                         Judge allocatedJudge) {
+        return hearingBookings.stream()
+            .map(element -> {
+                HearingBooking hearingBooking = element.getValue();
+
+                JudgeAndLegalAdvisor selectedJudge =
+                    getSelectedJudge(hearingBooking.getJudgeAndLegalAdvisor(), allocatedJudge);
+
+                removeAllocatedJudgeProperties(selectedJudge);
+
+                hearingBooking.setJudgeAndLegalAdvisor(selectedJudge);
+
+                return buildHearingBookingElement(element.getId(), hearingBooking);
+            }).collect(toList());
+    }
+
+    public List<Element<HearingBooking>> resetHearingJudge(List<Element<HearingBooking>> hearingBookings,
+                                                           Judge allocatedJudge) {
+        return hearingBookings.stream()
+            .map(element -> {
+                HearingBooking hearingBooking = element.getValue();
+                JudgeAndLegalAdvisor judgeAndLegalAdvisor = hearingBooking.getJudgeAndLegalAdvisor();
+
+                if (isNotEmpty(judgeAndLegalAdvisor)
+                    && allocatedJudge.equalsJudgeAndLegalAdvisor(judgeAndLegalAdvisor)) {
+                    return buildHearingBookingElement(element.getId(), resetJudgeDetails(hearingBooking));
+                }
+
+                return element;
+            }).collect(toList());
+    }
+
     private boolean isPastHearing(Element<HearingBooking> element) {
         return ofNullable(element.getValue())
             .map(HearingBooking::getStartDate)
             .filter(hearingDate -> hearingDate.isBefore(time.now()))
             .isPresent();
     }
+
+    private HearingBooking resetJudgeDetails(HearingBooking hearingBooking) {
```

Comment:
This process be simplified a bit; check my PR to see a method called prepareJudgeFields which will return a judge in the format we want; you could maybe use this

Reply:
As discussed, would rather use my current implementation to avoid rebuilding each HearingBooking in the collection unnecessarily.